File: backend/app/schemas/system.py
"""Safe operational state responses without exposing credentials."""

from pydantic import BaseModel


class SystemReadinessResponse(BaseModel):
    retrieval_mode: str
    external_model_calls_enabled: bool
    vector_search_enabled: bool
    chat_generation_enabled: bool
    reranker_enabled: bool
    web_search_enabled: bool
    web_search_provider: str
    web_search_monthly_request_limit: int
    web_search_max_requests_per_answer: int
    web_search_allowed_domains: list[str]
    missing_embedding_fields: list[str]
    missing_chat_fields: list[str]
    missing_reranker_fields: list[str]
    documents: int
    child_chunks: int
    catalog_artifacts: int
    media_assets: int
    approved_document_links: int = 0

File: frontend/src/types/system.ts
export interface SystemReadinessResponse {
  retrieval_mode: string;
  external_model_calls_enabled: boolean;
  vector_search_enabled: boolean;
  chat_generation_enabled: boolean;
  reranker_enabled: boolean;
  web_search_enabled: boolean;
  web_search_provider: string;
  web_search_monthly_request_limit: number;
  web_search_max_requests_per_answer: number;
  web_search_allowed_domains: string[];
  missing_embedding_fields: string[];
  missing_chat_fields: string[];
  missing_reranker_fields: string[];
  documents: number;
  child_chunks: number;
  catalog_artifacts: number;
  media_assets: number;
  approved_document_links: number;
}
